### DISCUSSION TITLE
Can't make .ENV work no matter what

### DISCUSSION THREAD

**Discussion by comunik16** [vercel/next.js#46418] [Help]
I need to make a cms headless website using next js (latest version) and wordpress (latest) for a job.  I'm trying for days to work around an issue with no success.

I've even cloned the ISR Blog Example Using Next.js and WordPress "npx create-next-app --example cms-wordpress cms-wordpress-app" , but i always get the same error: 

_Error - Failed to load next.config.js, see more info here https://nextjs.org/docs/messages/next-config-error
TypeError: Cannot read properties of null (reading '0')`._

If I type ``echo $WORDPRESS_API_URL`` on terminal I get an empty result. If I try to console ``process.env.WORDPRESS_API_URL`` I get undefined. 

_Even If I set ``const API_URL = "http://localhost/site_test/graphql`` it does not work._ 

At the wordpress side everything is ok.

What I'm missing? Thank you so much for your attention

----

.env.local: 

``
WORDPRESS_API_URL=http://localhost/site_test/graphql
``

My next.config.js:

``
if (!process.env.WORDPRESS_API_URL) {
  throw new Error(`
    Please provide a valid WordPress instance URL.
    Add to your environment variables WORDPRESS_API_URL.
  `);
}

/** @type {import('next').NextConfig} */
module.exports = {
  images: {
    domains: [
      process.env.WORDPRESS_API_URL.match(/(?!(w+)\.)\w*(?:\w+\.)+\w+/)[0], // Valid WP Image domain.
      "0.gravatar.com",
      "1.gravatar.com",
      "2.gravatar.com",
      "secure.gravatar.com",
    ],
  },
};
``

**Comment by ciruz** [comment]
Hey,
this sounds like the .env variables are not properly set up, or you have a different `NODE_ENV` set, maybe global somehow in your system?

Did you also try it with a global env file (`.env`) without any environment at all? 

Maybe you are just launching the application with a different environment and therefore it loads a different .env file, where your variable is really missing.

You can find more informations about the environment variable load order [here](https://nextjs.org/docs/basic-features/environment-variables#environment-variable-load-order).

Also try to print out the `NODE_ENV`, depends on your operating system.

**Comment by osdeibi** [comment]
I'm facing the same issue
